controllers/datafoundation.go

```diff
@@ -0,0 +1,113 @@
+package controllers
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+)
+
+type dataFoundationSpec struct {
+	usableCapacityInTiB     int
+	onboardingValidationKey string
+}
+type dataFoundationReconciler struct {
+	*ManagedFusionOfferingReconciler
+	spec                          dataFoundationSpec
+	onboardingValidationKeySecret *corev1.Secret
+}
+
+func dfSetupWatches(controllerBuilder *builder.Builder) {
+	controllerBuilder.
+		Owns(&corev1.Secret{})
+}
+
+func dfReconcile(managedFusionOfferingReconciler *ManagedFusionOfferingReconciler) error {
+	var r *dataFoundationReconciler
+	r.ManagedFusionOfferingReconciler = managedFusionOfferingReconciler
```

Comment:
This will throw a nil access error. You need to create a concrete `dataFoundationReconciler` before you can assign to any field including the `ManagedFusionOfferingReconciler ` embedded field. Also I see no reason to use a pointer here in the first place.

  ```suggestion
	r := dataFoundationReconciler{
		ManagedFusionOfferingReconciler: ManagedFusionOfferingReconciler
	}
```

But there is even a better way. You have an `r.initReconciler` function that you call just a couple of lines below. Why not use it to set up the embedded field, the same as you do for other fields.

Then this line can be as simple as 
```suggestion
	r := dataFoundationReconciler{
```

And the first line in `initReconcier` will be
```
	r.ManagedFusionOfferingReconciler = offeringReconciler
```

Assuming `offeringReconciler` will be the name of the input argument